library/ssl_msg.c

```diff
@@ -711,8 +709,7 @@ static void ssl_extract_add_data_from_record(unsigned char *add_data,
     memcpy(cur, rec->ver, sizeof(rec->ver));
     cur += sizeof(rec->ver);
 
-#if defined(MBEDTLS_SSL_DTLS_CONNECTION_ID) && \
-    MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT == 1
+#if defined(MBEDTLS_SSL_DTLS_CONNECTION_ID)
```

Comment:
This looks wrong: we're removing the case where `MBEDTLS_SSL_DTLS_CONNECTION_ID` is defined and `MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT` is 1. So the block of code from here to the `#elif` should be removed, and the `#elif` should become an `#if`.